pyomo/core/base/block.py

```diff
@@ -1028,6 +1044,12 @@ def find_component(self, label_or_component):
             cuid = ComponentUID(label_or_component)
         return cuid.find_component_on(self)
 
+    @contextmanager
+    def _declare_reserved_components(self):
+        self._Block_reserved_words = ()
+        yield
+        del self._Block_reserved_words
```

Comment:
Why doesn't this restore `_Block_reserved_words` to what it was before?

Reply:
It doesn't need to.  This is masking the class attribute with an instance attribute.  "Restoring" the class attribute just boils down to deleting the local instance attribute.

Reply:
Ohhh, that makes sense.